hooks/tk-flame_actions.py

```diff
@@ -247,114 +296,467 @@ def _import_batch_group_from_shot(self, sg_publish_data):
+
+    @property
+    def setup_path_template(self):
+        """
+        Template built from the "write_file_setup_path_template" entry from the configuration dictionary.
+
+        This template helps to setup the write_file_node by specifying where to write the setup
+
+        :return: Setup path template
+        :rtype: TemplatePath
+        """
+
+        template_name = self.configuration.get("write_file_setup_path_template")
+        return self.parent.sgtk.templates.get(template_name)
+
+    @property
+    def version_padding(self):
+        """
+        Padding to use on the version attribute on the Write File node.
+
+        This is defined by the "write_file_version_padding" entry of the configuration dictionary
+
+        :return: Write File version padding
+        :rtype: int
+        """
+
+        return self.configuration.get("write_file_version_padding")
+
+    @property
+    def frame_padding(self):
+        """
+        Padding to use on the frame attribute on the Write File node.
+
+        This is defined by the "write_file_frame_padding" entry of the configuration dictionary
+
+        :return: Write File frame padding
+        :rtype: int
+        """
+
+        return self.configuration.get("write_file_frame_padding")
+
+    ##############################################################################################################
+    # helper methods which can be subclassed in custom hooks to fine tune the behavior of things
+
+    def _create_batch_group(self, shot_info):
+        """
+        Create a new Batch group using the current Shot information
+
+        :param shot_info: Metadata of the current Shot
+        :type shot_info: dict
+        """
+        app = self.parent
+
+        app.log_debug("Creating the batch group using '%s'" % shot_info)
+
+        # Get the clips from sg_published_files
+        clips = self._get_clips_from_published_files(shot_info)
+
+        app.log_debug("Found clips %s" % clips)
+
+        if not clips:
+            raise FlameActionsError("No clip to load")
+
+        # Get the frame information of the Batch group from the sg_versions
+        start_frame, last_frame = self._extract_frame_range_from_version(shot_info)
+
+        # Start to build the Batch Group attribute dictionary
+        batch_group_info = {"name": shot_info["code"]}
+
+        # Add frame information to the batch_group dictionary if available
+        if None not in [start_frame, last_frame]:
+            batch_group_info["start_frame"] = start_frame
+            batch_group_info["duration"] = (int(last_frame) - int(start_frame)) + 1
+
+        # Create the Batch Group
+        flame.batch.go_to()
+
+        if not flame.batch.create_batch_group(**batch_group_info):
+            raise FlameActionsError("Unable to create a Batch Group")
+
+        have_write_file = False
+
+        # Import every clips!
+        for clip in clips:
+            if self._exists(clip["path"]):
```

Comment:
Same comment here, is it worth checking that file exist since it was returned from a list and will be checked again in flame code.